pkg/applicationmanager/controllers/applicationrestore.go

```diff
@@ -2152,3 +2176,30 @@ func (a *ApplicationRestoreController) cleanupResources(restore *storkapi.Applic
 	}
 	return nil
 }
+
+// filterOrphanedResourcesForVMRestore filters out resources such as PVC, ConfgiMap and Secrets that are not associated to any VM specified in the
+// includeResources. Such resources called orhpaned resources are removed from includeResrouce during restore for VM specific restore operation.
+func (a *ApplicationRestoreController) filterOrphanedResourcesForVMRestore(restore *storkapi.ApplicationRestore) ([]storkapi.ObjectInfo, error) {
+
+	objectInfo := make([]storkapi.ObjectInfo, 0)
+	backup, err := storkops.Instance().GetApplicationBackup(restore.Spec.BackupName, restore.Namespace)
+	if err != nil {
+		return nil, fmt.Errorf("error getting backup spec for VirtualMachine specific restore: %v", err)
+	}
+	// get objectMap of includeResources
+	objectMap := storkapi.CreateObjectsMap(restore.Spec.IncludeResources)
```

Comment:
To understand the flow, if there are 1000 Virtmachines, UI would read the backupObejct and fill the IncludeResoruce right? 
I am thinking can we optimzie this for VM flow. As UI sets `BackupObjectType` can we piggy back on this and once stork fetches the resoruce.json file and based on this flag, can we void using IncludeResources?
My thought was as we are making VM first class citizen, it would be good to have flow where minimal info is sent from UI to backed and stork has the intelligence to figure it out,

Reply:
We have split resources into application and volumes. Volumes restore happens first and resource next. There is no mapping built  for application to volumes in the current architecture.  Hence becomes extremely complex to build mapping once during Volume stage and again during application stage, even during application stage, you will end up building one for VM(self) and other for other core resources like configMap and secrets.
So includeResource seem to be the best bet in the current architecture, for which we can map the dependency and process.

Reply:
Discussed with @vsundarraj-px and got clear understanding. As only VirtMachine names are passed in IncludeResources, this approach is fine.